### PR TITLE
Add search_runs() and e2e testing framework

### PR DESCRIPTION
**Pull request by alkasm**

### CODE REVIEW

File: nominal/sdk.py [hunk 1]
```diff
@@ -366,8 +366,8 @@ def create_run(
         self,
         title: str,
         description: str,
-        start_time: datetime | IntegralNanosecondsUTC,
-        end_time: datetime | IntegralNanosecondsUTC,
+        start: datetime | IntegralNanosecondsUTC,
+        end: datetime | IntegralNanosecondsUTC,
```
Comment (alkasm) [inline nominal/sdk.py:370]: this is a breaking change, but ensures everything is uniform.

File: pyproject.toml [hunk 2]
```diff
@@ -57,3 +58,13 @@ strict = true
...
 module = "nominal._api.*"
 implicit_reexport = true
+
+[tool.pytest.ini_options]
+filterwarnings = [
+    "ignore:The 'strict' parameter is no longer needed on Python 3+."
+]
+markers = [
+    "e2e: marks tests as e2e, which require a token (deselect with '-m \"not e2e\"')",
+]
```
Comment (alkasm) [inline pyproject.toml:68]: i can get rid of the markers

**Review by alxhill — APPROVED** [review]
nice tests!

File: pyproject.toml [hunk 3]
```diff
@@ -57,3 +58,13 @@ strict = true
...
 module = "nominal._api.*"
 implicit_reexport = true
+
+[tool.pytest.ini_options]
+filterwarnings = [
+    # DeprecationWarning: The 'strict' parameter is no longer needed on Python 3+. This will raise an error in urllib3 v2.1.0.
+    # from conjure-python-client, in conjure_python_client/_http/requests_client.py,
+    # TransportAdapter.init_poolmanager() uses urllib3.poolmanager.PoolManager(..., strict=True) which is deprecated.
+    "ignore:The 'strict' parameter is no longer needed on Python 3+."
+]
```
Comment (alkasm) [inline pyproject.toml:68]: annoyingly, this warning pops up when the pool is instantiated, which is _not_ when the client is constructed, so I can't ignore this in the client library easily.

**Pull request merged**